api/strategies/googleStrategy.js

```diff
@@ -10,7 +17,104 @@ const getProfileDetails = (profile) => ({
+/**
+ * Retrieves the resource name for a Google Workspace group
+ * @param {string} accessToken - OAuth access token
+ * @param {string} groupEmail - Email address of the group
+ * @returns {Promise<string|null>} Group resource name or null if not found
+ */
+async function getGroupResourceName(accessToken, groupEmail) {
+  try {
+    const response = await axios.get(
+      'https://cloudidentity.googleapis.com/v1/groups:lookup',
+      {
+        params: {
+          'groupKey.id': groupEmail,
+        },
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    );
+    return response.data.name;
+  } catch (error) {
+    logger.error('[getGroupResourceName] Error looking up group:', error.response?.data || error);
+    return null;
+  }
+}
+
+/**
+ * Verifies if a user is a member of the specified Google Workspace group
+ * @param {string} accessToken - OAuth access token
+ * @param {string} userEmail - Email address of the user
+ * @returns {Promise<boolean>} True if user is a member or if no group is specified
+ */
+async function checkGroupMembership(accessToken, userEmail) {
+  try {
+    const allowedGroup = process.env.GOOGLE_WORKSPACE_GROUP;
+    if (!allowedGroup) {
+      return true;
+    }
+
+    const groupName = await getGroupResourceName(accessToken, allowedGroup);
+    if (!groupName) {
+      logger.error('[checkGroupMembership] Could not find group resource name');
+      return false;
+    }
+
+    const response = await axios.get(
+      `https://cloudidentity.googleapis.com/v1/${groupName}/memberships:checkTransitiveMembership`,
+      {
+        params: { query: `member_key_id == '${userEmail}'` },
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    );
+
+    return response.data.hasMembership || false;
+  } catch (error) {
+    logger.error(
+      '[checkGroupMembership] Error checking group membership:',
+      { userEmail, error: error.response?.data || error }
+    );
+    return false;
+  }
+}
+
+/**
+ * Handles Google OAuth login process with group membership verification
+ * @param {string} accessToken - OAuth access token
+ * @param {string} refreshToken - OAuth refresh token
+ * @param {Object} profile - User profile from Google
+ * @param {Function} done - Passport callback function
+ */
+async function googleLogin(accessToken, refreshToken, profile, done) {
+  try {
+    const userEmail = profile.emails[0].value;
+    const isMember = await checkGroupMembership(accessToken, userEmail);
+
+    if (!isMember) {
+      return done(null, false);
+    }
+
+    const socialLoginCallback = (err, user) => {
+      if (err) {
+        return done(err);
+      }
+      done(null, user);
+    };
+
+    return socialLogin('google', getProfileDetails)(
+      accessToken,
+      refreshToken,
+      profile,
+      socialLoginCallback
+    );
+  } catch (error) {
+    return done(error);
+  }
+}
```

Comment:
thank you for doing this manual. but would it be possible to use the google's official `npm` package

https://googleapis.dev/nodejs/googleapis/latest/cloudidentity/classes/Cloudidentity.html
https://www.npmjs.com/package/@googleapis/cloudidentity